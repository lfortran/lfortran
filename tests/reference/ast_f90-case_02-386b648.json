{
    "basename": "ast_f90-case_02-386b648",
    "cmd": "lfortran --show-ast-f90 --no-indent --no-color {infile}",
    "infile": "tests/../integration_tests/case_02.f90",
    "infile_hash": "76d3020fba89787d565870ed03c1b5e8914d12b062cb9ca7a8c37550",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_f90-case_02-386b648.stdout",
    "stdout_hash": "287601388e018816a37bd53d489512a90322f78a6e2c91eee795bd68",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}