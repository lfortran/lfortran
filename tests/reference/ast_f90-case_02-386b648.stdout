program case_02
implicit none

! local variable declaration
integer :: marks, out
marks = 81

select case (marks)

    case (91:100)
        out = 0
        print *, "Excellent!"

    case (81:90)
        out = 1
        print *, "Very good!"

    case (71:80)
        out = 2
        print *, "Well done!"

    case (61:70)
        out = 3
        print *, "Not bad!"

    case (41:60)
        out = 4
        print *, "You passed!"

    case (:40)
        out = 5
        print *, "Better try again!"

    case default
        out = 6
        print *, "Invalid marks"

end select
print *, "Your marks are ", marks

if (out /= 1) then
    error stop
end if

! case default at the top
select case (marks)

    case default
        print *, "Invalid marks"

    case (91:100)
        print *, "Excellent!"

    case (81:90)
        print *, "Very good!"

    case (71:80)
        print *, "Well done!"

    case (61:70)
        print *, "Not bad!"

    case (41:60)
        print *, "You passed!"

    case (:40)
        print *, "Better try again!"

end select
print *, "Your marks are ", marks

! case default in the middle
select case (marks)

    case (91:100)
        print *, "Excellent!"

    case (81:90)
        print *, "Very good!"

    case default
        print *, "Invalid marks"

    case (71:80)
        print *, "Well done!"

    case (61:70)
        print *, "Not bad!"

    case (41:60)
        print *, "You passed!"

    case (:40)
        print *, "Better try again!"

end select
print *, "Your marks are ", marks

end program case_02
